File: leetsync/1063-best-sightseeing-pair/best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int n = values.length;
        int result = 0;
        int maxIValue = values[0];
        for (int j = 1; j < n; j++) {
            int jValue = values[j] - j;
            result = Math.max(result, maxIValue + jValue);
            maxIValue = Math.max(maxIValue, values[j] + j);
        }
        return result;
    }
}

File: leetsync/1063-best-sightseeing-pair/best-sightseeing-pair.go
func maxScoreSightseeingPair(values []int) int {
	n := len(values)
	result := 0
	x := values[0]
	for j := 1; j < n; j++ {
		y := values[j] - j
		result = max(result, x+y)
		x = max(x, values[j]+j)
	}
	return result
}